apps/wallet/frontend/src/components/ListAllocationRequests.tsx

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2024 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React from 'react';
+import { Card, CardContent, Chip, Stack } from '@mui/material';
+import Typography from '@mui/material/Typography';
+import { useTokenStandardAllocationRequests } from '../hooks/useTokenStandardAllocationRequests';
+import { AmountDisplay, Loading } from '@lfdecentralizedtrust/splice-common-frontend';
+import { AllocationRequest } from '@daml.js/splice-api-token-transfer-instruction/lib/Splice/Api/Token/AllocationRequestV1/module';
+import { Contract } from '@lfdecentralizedtrust/splice-common-frontend-utils';
+import { usePrimaryParty } from '../hooks';
+import dayjs from 'dayjs';
+import relativeTime from 'dayjs/plugin/relativeTime';
+import BigNumber from 'bignumber.js';
+import BftAnsEntry from './BftAnsEntry';
+import { ArrowCircleLeftOutlined } from '@mui/icons-material';
+
+dayjs.extend(relativeTime);
+
+const ListAllocationRequests: React.FC = () => {
+  const allocationRequestsQuery = useTokenStandardAllocationRequests();
+  const primaryPartyId = usePrimaryParty();
+
+  if (allocationRequestsQuery.isLoading || !primaryPartyId) {
+    return <Loading />;
+  }
+  if (allocationRequestsQuery.isError) {
+    return (
+      <Typography color="error">
+        Error loading allocation requests: {JSON.stringify(allocationRequestsQuery.error)}
+      </Typography>
+    );
+  }
+
+  const allocationRequests = allocationRequestsQuery.data || [];
+
+  return (
+    <Stack spacing={4} direction="column" justifyContent="center" id="allocation-requests">
+      <Typography mt={6} variant="h4">
+        Allocation Requests <Chip label={allocationRequests.length} color="success" />
+      </Typography>
+      {allocationRequests.map(ar => (
+        <AllocationRequestDisplay key={ar.contractId} request={ar} userParty={primaryPartyId} />
+      ))}
+    </Stack>
+  );
+};
+
+const AllocationRequestDisplay: React.FC<{
+  request: Contract<AllocationRequest>;
+  userParty: string;
+}> = ({ request, userParty }) => {
+  const { transferLegs, settlement } = request.payload;
+
+  const transferLegId = Object.keys(transferLegs).find(
+    transferLegId => transferLegs[transferLegId].sender === userParty
+  );
+  const transferLeg = transferLegs[transferLegId || ''];
+  if (!transferLeg) return null;
+
+  const { amount, receiver } = transferLeg;
+  const { allocateBefore, settlementRef, executor } = settlement;
+  const id = [settlementRef.id, settlementRef.cid].filter(x => !!x).join(' - ');
```

Comment:
This join has some security issues, as an attacker could also inline `-<some-other-contract-id>` in their `id` field.

I'd suggest to just make it two separate labelled text fields.